pkgs/tools/misc/android-tools/default.nix

```diff
@@ -0,0 +1,39 @@
+{ lib, stdenv, fetchurl
+, cmake, perl, go
+, protobuf, zlib, gtest, brotli, lz4, zstd, libusb1, pcre2
+}:
+
+stdenv.mkDerivation rec {
+  pname = "android-tools";
+  version = "31.0.0p1";
+
+  src = fetchurl {
```

Comment:
Could this be using `fetchFromGitHub`?

Reply:
We need to use release tarball as the repo only have build-system and documentation. Upstream doesn't vendorize platform-tools source code in repo. They use CPack to create tarball from only used source files. 
https://github.com/nmeum/android-tools/tree/master/vendor

Reply:
How many sources would we need to declare in Nix to achieve the same? I'd prefer if all sources were declared transparently instead of relying on an opaque tarball.

Reply:
Well in that case we need to checkout some of AOSP repositories. Those repositories are actually HUGE and Google doesn't provide snapshots from tags. Checking out repositories takes nearly 1 hour on my computer.

Reply:
The whole tree is obviously huge but we shouldn't need all of it. That's why I was asking.

Reply:
We can declare the source files but there is 13-14 repos and lots of files all aroud the tree. Leaving the tracking job to CPack would be much easier. I don't agree tarballs are opaque as you can reproduce them. We can create our tarballs or vendorize CPack output in a repository.

Reply:
Is it legally OK if we mirror some of Google's repositories on other platforms like [this one](https://github.com/musfay/adb).

Reply:
> there is 13-14 repos

Yeah, that's a bit much.

> I don't agree tarballs are opaque as you can reproduce them. 

The problem with that is that the reproduction is not transparent and in this case probably not even reproducible at a different point in time.  
You don't know anything about how CPack was made nor which version of the sources it fetched.    
Google's repo doesn't have a single "revision" or an equivalent; it's an aggregation of whatever happens to be at the top of a few hundred repos at a certain point in time.      
It'd only become transparent if the generation was declared in Nix because then you'd have to lock the exact revision and hash of every included repo.

As I said though, too many repos. I was hoping for, like, 5 which would've been a feasible amount to maintain ourselves.

> Is it legally OK if we mirror some of Google's repositories

The APL2 allows you to redistribute source code in any way you like but, why?

Reply:
> > I don't agree tarballs are opaque as you can reproduce them.
> 
> The problem with that is that the reproduction is not transparent and in this case probably not even reproducible at a different point in time.
> You don't know anything about how CPack was made nor which version of the sources it fetched.
> Google's repo doesn't have a single "revision" or an equivalent; it's an aggregation of whatever happens to be at the top of a few hundred repos at a certain point in time.
> It'd only become transparent if the generation was declared in Nix because then you'd have to lock the exact revision and hash of every included repo.

It makes sense now. Thanks for the definition!

> As I said though, too many repos. I was hoping for, like, 5 which would've been a feasible amount to maintain ourselves.
Yep there is lots of repos. 
> > Is it legally OK if we mirror some of Google's repositories
> 
> The APL2 allows you to redistribute source code in any way you like but, why?
Just asked, offtopic.